src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -499,6 +499,28 @@ public override void Write(byte[] buffer, int offset, int count)
             WriteAsyncCore(buffer.AsMemory(offset, count), default, useAsync: false).AsTask().GetAwaiter().GetResult();
         }
 
+        /// <inheritdoc/>
+        public override void Write(ReadOnlySpan<byte> buffer)
+        {
+            // Logically this is doing the same thing as the base Stream, however CryptoStream clears arrays before
+            // returning them to the pool, whereas the base Stream does not.
+            byte[] sharedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
+
+            try
+            {
+                buffer.CopyTo(sharedBuffer);
+
+                // We want to keep calling the virtual Write(byte[]...) so that derived CryptoStream types continue
+                // to get the array overload called from the span one.
+                Write(sharedBuffer, 0, buffer.Length);
+            }
+            finally
+            {
+                CryptographicOperations.ZeroMemory(sharedBuffer.AsSpan(0, buffer.Length));
+                ArrayPool<byte>.Shared.Return(sharedBuffer);
```

Comment:
Can this use the [`CryptoPool`](https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/Common/src/System/Security/Cryptography/CryptoPool.cs#L9) instead the ArrayPool?
It does the same thing, but abstracted away via Rent/Return.